### ISSUE TITLE
Version 6.6.0 has incorrect imports for lodash/isEqualWith.

### ISSUE THREAD

**Issue by adriantrunzo** [testing-library/jest-dom#641]
- `@testing-library/jest-dom` version: 6.6.0
- `node` version: 20.17.0
- `jest` (or `vitest`) version: vitest 2.1.2
- `npm` (or `yarn`) version: pnpm 9.12

### Relevant code or config:

The `node_modules/@testing-library/jest-dom/dist/vitest.mjs` file currently reads:

```ts
import { expect } from 'vitest';
import { e as extensions } from './matchers-12582698.mjs';
import 'redent';
import '@adobe/css-tools';
import 'dom-accessibility-api';
import 'aria-query';
import 'chalk';
import 'lodash/isEqualWith.js';
import 'css.escape';
import 'lodash/isEqualWith';

/* istanbul ignore file */


expect.extend(extensions);
```

All other `dist` files have the same problem.

Notice the two different imports of `lodash/isEqualWith`. The second one is incorrect and was added in #637 : https://github.com/testing-library/jest-dom/pull/637/files#diff-a136cc1088d17955bf3e51c2cb464c75890d15a5a008c05a9d17c68609b0a2f5R1.

See also: https://github.com/search?q=repo%3Atesting-library%2Fjest-dom%20lodash%2FisEqualWith&type=code

### What you did:

Try to upgrade to 6.6.0.

### What happened:

```
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Suites 8 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

Error: Cannot find module '/Users/.../node_modules/.pnpm/@testing-library+jest-dom@6.6.0/node_modules/lodash/isEqualWith' imported from /Users/.../node_modules/.pnpm/@testing-library+jest-dom@6.6.0/node_modules/@testing-library/jest-dom/dist/vitest.mjs
Did you mean to import "lodash/isEqualWith.js"?
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/8]
```

### Problem description:

Incorrect import.

### Suggested solution:

Change the import of lodash/isEqualWith in [src/to-have-selection.js](https://github.com/testing-library/jest-dom/blob/9b148043d082a83f0ae5cdc03cdfc6a7c4573e6e/src/to-have-selection.js#L1)

**Comment by gnapse** [comment]
Hmmm, I wonder why wasn't this caught in the tests and CI checks. Anyway. Will fix it.

**Comment by gnapse** [comment]
Wait, there's something weird going on here. We've had the same import for some time, in [src/to-have-value.js](https://github.com/testing-library/jest-dom/blob/9b148043d082a83f0ae5cdc03cdfc6a7c4573e6e/src/to-have-value.js#L1).

https://github.com/testing-library/jest-dom/blob/9b148043d082a83f0ae5cdc03cdfc6a7c4573e6e/src/to-have-value.js#L1

Looking at the difference with the new one, the new one lacks the extension.

https://github.com/testing-library/jest-dom/blob/9b148043d082a83f0ae5cdc03cdfc6a7c4573e6e/src/to-have-selection.js#L1

Ugh, I hate it with the extension is needed in imports. I wonder if that's why it worked in CI and not for you.

**Comment by ddolcimascolo** [comment]
Was coming here for this exact same issue :) 

And I don't know if you're aware but this [happened before](https://github.com/testing-library/jest-dom/issues/598) so this might require an eslint rule or something.

Thanks for your work on this library!

Cheers,
David

**Comment by qeleb** [comment]
+1

**Comment by gnapse** [comment]
The fix is now released in v6.6.1.
